Package.swift

```diff
@@ -311,3 +311,18 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(path: "../swift-driver"),
     ]
 }
+
+// Because SwiftSyntax is closely tied to the compiler, only attempt to build
+// the package editor library if we're in a build-script environment and can
+// assume we're using a just-built compiler and SwiftSyntax library.
+if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] != nil {
```

Comment:
Perhaps we should have some additional control over whether to require `swift-syntax` and a matching compiler.  `SWIFTCI_USE_LOCAL_DEPS` is the same environment variable that controls whether to fetch the other dependencies or expect to find them, and it would be good to have independent control over whether to try to build the package editor.

Reply:
👍 I updated this to use `SPM_ENABLE_PACKAGE_EDITOR` instead, and the bootstrap script passes that for the stage 2 build. We now also define ENABLE_PACKAGE_EDITOR when building the `Commands` module, and I'm conditionally compiling the new subcommands there instead of breaking them out into a separate executable `swift-package` would need to forward to. This setup should keep the build relatively simple as long as we're ok not supporting this functionality in stage 1 CMake builds, which IMO is fine (and then we don't need to setup CMake for SwiftSyntax either).